src/course-unit/move-modal/moveModal.test.tsx

```diff
@@ -0,0 +1,191 @@
+
+    expect(getByText(messages.moveModalTitle.defaultMessage.replace(' {displayName}', ''))).toBeInTheDocument();
+    expect(within(breadcrumbs).getByText(messages.moveModalBreadcrumbsBaseCategory.defaultMessage)).toBeInTheDocument();
+    expect(
+      within(categoryIndicator).getByText(messages.moveModalBreadcrumbsSections.defaultMessage),
+    ).toBeInTheDocument();
+    expect(getByRole('button', { name: messages.moveModalSubmitButton.defaultMessage })).toBeInTheDocument();
+    expect(getByRole('button', { name: messages.moveModalCancelButton.defaultMessage })).toBeInTheDocument();
+  });
+
+  it('correctly navigates through the structure list', async () => {
+    const { getByText, getByRole, getByTestId } = renderComponent();
+    const breadcrumbs: HTMLElement = getByTestId('move-xblock-modal-breadcrumbs');
+    const categoryIndicator: HTMLElement = getByTestId('move-xblock-modal-category');
+
+    expect(within(breadcrumbs).getByText(messages.moveModalBreadcrumbsBaseCategory.defaultMessage)).toBeInTheDocument();
+    expect(
+      within(categoryIndicator).getByText(messages.moveModalBreadcrumbsSections.defaultMessage),
+    ).toBeInTheDocument();
+    sections.map((section) => (
+      expect(getByText(section.display_name)).toBeInTheDocument()
+    ));
+
+    await waitFor(() => userEvent.click(getByRole('button', { name: new RegExp(sections[1].display_name, 'i') })));
+    await waitFor(() => {
+      expect(
+        within(categoryIndicator).getByText(messages.moveModalBreadcrumbsSubsections.defaultMessage),
+      ).toBeInTheDocument();
+      expect(within(breadcrumbs).getByText(sections[1].display_name)).toBeInTheDocument();
+      subsections.map((subsection) => (
+        expect(getByRole('button', { name: new RegExp(subsection.display_name, 'i') })).toBeInTheDocument()
+      ));
+    });
+
+    await waitFor(() => userEvent.click(getByRole('button', { name: new RegExp(subsections[1].display_name, 'i') })));
+    await waitFor(() => {
+      expect(
+        within(categoryIndicator).getByText(messages.moveModalBreadcrumbsUnits.defaultMessage),
+      ).toBeInTheDocument();
+      expect(within(breadcrumbs).getByText(subsections[1].display_name)).toBeInTheDocument();
+      units.map((unit) => (
+        expect(getByRole('button', { name: new RegExp(unit.display_name, 'i') })).toBeInTheDocument()
+      ));
+    });
+
+    await waitFor(() => userEvent.click(getByRole('button', { name: new RegExp(units[0].display_name, 'i') })));
+    await waitFor(() => {
+      expect(
+        within(categoryIndicator).getByText(messages.moveModalBreadcrumbsComponents.defaultMessage),
+      ).toBeInTheDocument();
+      expect(within(breadcrumbs).getByText(units[0].display_name)).toBeInTheDocument();
+      components.forEach((component) => {
+        if (component.display_name) {
+          expect(getByText(component.display_name)).toBeInTheDocument();
+        }
+      });
+    });
+  });
```

Comment:
[optional]: I tried locally to remove some `waitFor` and unnecessary types and it works. Please check if you need to use waitFor so often in other tests in this file.

```suggestion
  it('correctly navigates through the structure list', async () => {
    const { getByText, getByRole, getByTestId } = renderComponent();
    const breadcrumbs = getByTestId('move-xblock-modal-breadcrumbs');
    const categoryIndicator = getByTestId('move-xblock-modal-category');

    expect(within(breadcrumbs).getByText(messages.moveModalBreadcrumbsBaseCategory.defaultMessage)).toBeInTheDocument();
    expect(
      within(categoryIndicator).getByText(messages.moveModalBreadcrumbsSections.defaultMessage)
    ).toBeInTheDocument();

    sections.forEach((section) => {
      expect(getByText(section.display_name)).toBeInTheDocument();
    });

    userEvent.click(getByRole('button', { name: new RegExp(sections[1].display_name, 'i') }));

    await waitFor(() => {
      expect(
        within(categoryIndicator).getByText(messages.moveModalBreadcrumbsSubsections.defaultMessage)
      ).toBeInTheDocument();
      expect(within(breadcrumbs).getByText(sections[1].display_name)).toBeInTheDocument();

      subsections.forEach((subsection) => {
        expect(getByRole('button', { name: new RegExp(subsection.display_name, 'i') })).toBeInTheDocument();
      });
    });

    userEvent.click(getByRole('button', { name: new RegExp(subsections[1].display_name, 'i') }));

    await waitFor(() => {
      expect(
        within(categoryIndicator).getByText(messages.moveModalBreadcrumbsUnits.defaultMessage)
      ).toBeInTheDocument();
      expect(within(breadcrumbs).getByText(subsections[1].display_name)).toBeInTheDocument();

      units.forEach((unit) => {
        expect(getByRole('button', { name: new RegExp(unit.display_name, 'i') })).toBeInTheDocument();
      });
    });

    userEvent.click(getByRole('button', { name: new RegExp(units[0].display_name, 'i') }));

    await waitFor(() => {
      expect(
        within(categoryIndicator).getByText(messages.moveModalBreadcrumbsComponents.defaultMessage)
      ).toBeInTheDocument();
      expect(within(breadcrumbs).getByText(units[0].display_name)).toBeInTheDocument();

      components.forEach((component) => {
        if (component.display_name) {
          expect(getByText(component.display_name)).toBeInTheDocument();
        }
      });
    });
  });
```

Reply:
Changed